src/include/mutex.h

```diff
@@ -30,8 +30,9 @@ pthread_mutex_lock(&NAME##_lock);
 
 #elif defined(_WIN32) // building for windows
 
-#define _WIN32_WINNT 0x0600
-#include <synchapi.h>
+#define _WIN32_WINNT 0x0600 // minimum target of Windows Vista
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
```

Comment:
This isn't a blocker, but why are we pulling in the extra stuff from windows.h since synchapi.h is a subset of that

Reply:
MSDN documents that you should use Windows.h and not the specific header.  Additionally, there are missing declarations which are required (e.g. `_AMD64_`, `_ARM64_` or `_X86_` which are conditional).

Reply:
This, basically. When we're just including `<synchapi.h>`, one of the headers that _that_ includes needs the architecture-identifying symbols that are only defined in `<windows.h>`. That was the problem i ran into when building swift-cmark with CMake instead of SwiftPM on Windows. The best we can do is restrict which headers and symbols get included with things like `WIN32_LEAN_AND_MEAN` or various `NO{subset}` symbols.

Reply:
That makes sense. I didn't look into it that deeply and at the end of day including a bit more stuff than we need to really isn't a big deal since it doesn't affect the build product.